scrapy/exporters.py

```diff
@@ -128,33 +129,52 @@ def __init__(self, file, **kwargs):
             self.encoding = 'utf-8'
         self.xg = XMLGenerator(file, encoding=self.encoding)
 
+    def _beautify_newline(self):
+        if self.indent_width:
```

Comment:
Actually, this current implementation is consistent for JSON and XML in the sense that `None==0`, because of this line: https://github.com/scrapy/scrapy/pull/2456/files#diff-b40c5755a5dda2b02aaac6bf4c8e2ff8R175.
However, I think you have a point and it would be even better to enforce the difference between `None` and `0`. I'm  working on it, I'll update this PR shortly 👍